eduNEXT/eox-nelp
# Jlc/gating completion first approx

## Description

Add progress and course grade event receiver to send progress_enrollmet data to futurex.

Helpful information to include:

- eox-core and social-auth-app-django added as dependencies.
- signals module created to manage receivers and tasks.
- Tests developed for receivers and tasks.
- The endpoint related to the enrollment progress is in this document on page 14.
https://drive.google.com/file/d/16PXRFCFZuRQMjleQqkm_qSKSfddpwT8g/view
- The data sent is mapped with this form.
![image](https://user-images.githubusercontent.com/51926076/227663200-83379100-4817-49dd-926f-7b2bd376aca7.png)
- Already deployed in stage.

## Testing instructions
Clone the repo. Complete a unit and check logs in sync or async containers. Depends on `CELERY_ALWAYS_EAGER = False`

Configure in settings:
```
    FUTUREX_API_URL = "example"
    FUTUREX_API_CLIENT_ID = "example""
    FUTUREX_API_CLIENT_SECRET = "example"
```
### After
**stage_test**:
![Peek 2023-03-24 18-15](https://user-images.githubusercontent.com/51926076/227660839-7b4c9225-24b7-4469-95d0-c671d4909ab9.gif)

## Additional information

Include anything else that will help reviewers and consumers understand the change.
- Jira story: 
https://edunext.atlassian.net/browse/FUTUREX-371

## Checklist for Merge
